pkg/logwrite/logwrite.go

```diff
@@ -100,19 +109,71 @@ func (l *LogFile) Close() error {
 	return l.File.Close()
 }
 
-// Rotate closes the current log file, rotates the files and creates an empty log file.
-func (l *LogFile) Rotate(maxLogFiles int) error {
+// Compress the given LogFile
+func (l *LogFile) Compress() error {
+	originalLogFile, err := os.OpenFile(l.Path, os.O_RDONLY, 0644)
+
+	if err != nil {
+		return err
+	}
+
+	// calculate the buffer size for rawfile
+	info, _ := originalLogFile.Stat()
+
+	var logFileSize = info.Size()
+	rawBytes := make([]byte, logFileSize)
+
+	// Read the log file into memory
+	inputBuffer := bufio.NewReader(originalLogFile)
+	_, err = inputBuffer.Read(rawBytes)
+
+	originalLogFile.Close()
+
+	if err != nil {
+		return err
+	}
+
+	// Create a GZIP writer
+	var outputBuffer bytes.Buffer
+	writer := gzip.NewWriter(&outputBuffer)
+	writer.Write(rawBytes)
+	writer.Close()
```

Comment:
I would recommend (if possible)
- use the simpler `os.Create` and `os.Open` functions rather than `os.OpenFile`;
- compress in a streaming fashion to avoid excessive buffering; which requires
- compressing to a new file (which you can rename afterwards if you want).

Maybe something like:
```go
input, err := os.Open(l.Path)
if err != nil {
    return err
}
defer input.Close()

output, err := os.Create(l.Path + gzipFileExtension)
if err != nil {
    return err
}
defer output.Close()

gzipWriter := gzip.NewWriter(output)
if _, err := io.Copy(gzipWriter, input); err != nil {
    return err
}
if err := gzipWriter.Close(); err != nil {
    return err
}
// delete the original?
if err := os.Remove(l.Path); err != nil {
    return err
}
// now we only have l.Path + gzipFileExtension
return nil
```

Reply:
Thanks @djs55, I will look into implementing that suggestion shortly. Its been a hell of a week. That suggestion sounds reasonable though. Worst case, Monday.

Reply:
apologies, I haven't forgotten about this, just been extremely busy. Hopefully should have a bit of time soon.